test/e2e/e2e_suite_test.go

```diff
@@ -30,37 +30,79 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/openshift-kni/mixed-cpu-node-plugin/test/e2e/infrastructure"
+	securityv1 "github.com/openshift/api/security/v1"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 )
 
+const defaultNamespaceName = "e2e-mixed-cpu-node-plugin"
+
 type TestFixture struct {
-	ctx    context.Context
+	Ctx    context.Context
 	Cli    client.Client
 	K8SCli *kubernetes.Clientset
-	NS     *corev1.Namespace
+	// TestingNS randomly generated for each test and gets deleted when the test ends
+	TestingNS *corev1.Namespace
 }
 
 var fixture TestFixture
 
 func TestE2e(t *testing.T) {
 	BeforeSuite(func() {
-		fixture.ctx = context.Background()
+		fixture.Ctx = context.Background()
 		Expect(initClient()).ToNot(HaveOccurred())
 		Expect(initK8SClient()).ToNot(HaveOccurred())
+		Expect(infrastructure.Setup(fixture.Ctx, fixture.Cli, GetNamespaceName())).ToNot(HaveOccurred())
```

Comment:
you may want to start early and annotate expectations with more descriptive messages on error.

Reply:
What do you mean by start early?
Actually the `Setup` function is very descriptive and tells exactly what went wrong.

Reply:
start early in the development cycle, piking up this habit. If the error message is descriptive enough I guess it's fine.

Reply:
I added another message just to be on the safe side